### ISSUE TITLE
Document lack of InputEvents with Input.action_press() and input.action_release()

### ISSUE THREAD

**Issue by GalanCM** [godotengine/godot#6135]
Mac OS X 10.11.6
Godot 2.1

Input.action_press() and Input.action_release() don't seem to do anything.

After a good chunk of trial and error I've simplified the code down to the following, attached to a global:

``` gdscript

extends Node

func _ready():
    set_process_input(true)
    print('trigger ui_accept')
    Input.action_press("ui_accept")

func _input(event):
    if event.is_action_pressed("ui_accept"):
        print('ui_accepted')
```

Which prints out:

```
trigger ui_accept
```

**Comment by eska014** [comment]
`Input.action_press()` and `Input.action_release()` do not send input events, they only change the action's state in the `Input` singleton.

If you want to keep using those functions, use `Input.is_action_pressed()` to check instead.
If you want input events, you can create your own InputEvent and feed it to the main loop:
```
var event = InputEvent()
event.set_as_action( 'ui_accept', true ) # second parameter specifies if pressed
get_tree().input_event( event )
```

**Comment by GalanCM** [comment]
Ah, thanks. I never would have guessed that's how they worked. In fact, I'd assumed that changing the Input singleton would fire off an InputEvent.

**Comment by Keyaku** [comment]
I also was initially confused, but learned how it worked after I looked at the code. So, you're not alone!

Do you mind closing this, though?

**Comment by eska014** [comment]
Let's keep it open as a documentation issue

**Comment by akien-mga** [comment]
IMO `Input.action_press` and `Input.action_release` should be deprecated and eventually removed. I don't see any use case for them, and their name is super confusing.

**Comment by Fr0sk** [comment]
> IMO `Input.action_press` and `Input.action_release` should be deprecated and eventually removed. I don't see any use case for them, and their name is super confusing.

While I agree, is there another way to set the action strength?

**Comment by KoBeWi** [comment]
Resolved in #31445
